packages/devtools_app/lib/src/screens/profiler/panes/method_table/method_table_model.dart

```diff
@@ -12,63 +14,136 @@ class MethodTableGraphNode extends GraphNode {
   MethodTableGraphNode({
     required this.name,
     required this.packageUri,
+    required this.sourceLine,
+    required int totalCount,
+    required int selfCount,
     required this.profileMetaData,
-  });
+    required this.stackFrameIds,
+  })  : _totalCount = totalCount,
+        _selfCount = selfCount,
+        _sourceUri = uriWithSourceLine(packageUri, sourceLine);
 
   factory MethodTableGraphNode.fromStackFrame(CpuStackFrame frame) {
     return MethodTableGraphNode(
       name: frame.name,
       packageUri: frame.packageUri,
+      sourceLine: frame.sourceLine,
+      totalCount: frame.inclusiveSampleCount,
+      selfCount: frame.exclusiveSampleCount,
       profileMetaData: frame.profileMetaData,
+      stackFrameIds: {frame.id},
     );
   }
 
   final String name;
 
   final String packageUri;
 
-  final ProfileMetaData profileMetaData;
-
-  String get id => '$name-$packageUri';
+  final int? sourceLine;
 
-  String get display => '$name ($packageUri)';
+  final String _sourceUri;
 
-  // TODO(kenz): implement the calculation for exclusive and inclusive count.
+  final ProfileMetaData profileMetaData;
 
-  /// The number of cpu samples where this frame is on top of the stack.
-  ///
-  /// This count is used to calculate self time.
-  int exclusiveSampleCount = 0;
+  /// The set of all [CpuStackFrame.id]s that contribute to this method table
+  /// node for a profile.
+  final Set<String> stackFrameIds;
 
-  /// The number of cpu samples where this frame is anywhere on the stack.
-  ///
-  /// This count is used to calculate total time.
-  int inclusiveSampleCount = 0;
+  String get id => '$name-$_sourceUri';
 
-  late double totalTimeRatio =
-      safeDivide(inclusiveSampleCount, profileMetaData.sampleCount);
+  String get display =>
+      '$name${_sourceUri.isNotEmpty ? ' - ($_sourceUri)' : ''}';
 
-  late Duration totalTime = Duration(
-    microseconds:
-        (totalTimeRatio * profileMetaData.time!.duration.inMicroseconds)
-            .round(),
-  );
+  /// The number of cpu samples where this method is on top of the stack.
+  int get selfCount => _selfCount;
```

Comment:
Can this just be `final int selfCount`? Ditto for `totalCount`.

Reply:
_selfCount and _totalCount are non-final because they are modified in the `merge` method.